### ISSUE TITLE
It should be possible to run an Open Source version of Automatisch

### ISSUE THREAD

**Issue by mjkaye** [automatisch/automatisch#2245]
It should be possible to run an AGPL version of Automatisch in self-hosted mode, without the EE code-base.

Code that is subject to the Enterprise licence is tightly integrated into the project. This makes it impossible to run Automatisch without the non-Open-Source code.

One example of this tight integration is `sentry.ee.js`. This is imported by `worker.js` and, so, Automatisch cannot be run without this Enterprise code.

As an Open Source project, the expectation is to have build/run options that allow installing only Open Source code, and for the Enterprise code to be in the form of add-ons.

Thank you for making this project.

**Comment by easyservices** [comment]
Agreed! Please make it so to respect the Open Source mindset and attract companies to your billable version if they are happy with the Open Source one as they will mostly go for an extended support to have cyber security patches. Thanks!

**Comment by Beanamumba** [comment]
Great idea!

**Comment by farukaydin** [comment]
Thanks for the feedback @mjkaye, @easyservices and @Beanamumba 

Keeping open-source and non-open-source parts together makes much easier for us to develop Automatisch further. I see the confusion here but it's okay to keep and run the software as is. Non-open-source parts might be loaded in your installation but will not be actively used in terms of functionality unless you have an Enterprise License.

**Comment by mjkaye** [comment]
I believe it is crucial that projects labeled as Open Source adhere to the principles and freedoms they claim to uphold.

The primary concern here is that the current implementation of `Automatisch` tightly integrates Enterprise components with its Open Source codebase, making it impossible to run a fully Open Source version of the project without including proprietary or licensed code. This violates the expectations set by the AGPL license, which requires users to have full access and freedom to modify the software.

### Specific Examples

A quick glance at the code reveals the following examples of proprietary code being included and called from the AGPL code. I'm sure there are many others:

https://github.com/automatisch/automatisch/blob/e686d3b0672e9254170b29ee4c97bd3fb2a165ef/packages/backend/src/workers/worker.js#L20

https://github.com/automatisch/automatisch/blob/e686d3b0672e9254170b29ee4c97bd3fb2a165ef/packages/backend/src/worker.js#L4

`sentry.ee.js` is imported by `worker.js`, preventing users from running a pure Open Source version of Automatisch. Here are some specific points that highlight why this integration is problematic:

1. **Tight Integration**: The fact that Enterprise components like `sentry.ee.js` are called directly from AGPL-licensed code (such as `worker.js`) means that running the Community Edition of `Automatisch` involves executing non-open-source code.
2. **Lack of Freedom**: Users who wish to use an Open Source version of Automatisch should be able to do so without being constrained by proprietary or licensed components. The current implementation does not provide this freedom, as users *must* run closed-source code for which they are not afforded the freedom to modify.

### Expected Behavior

As an Open Source project, `Automatisch` should offer build and run options that allow users to install only the AGPL-licensed (open-source) parts. If additional functionality is needed, it should be provided through add-ons or extensions that do not require modification of the core codebase.

### Suggested Steps Forward

To address this issue, you could follow these steps:

1. **Isolation**: Ensure that Enterprise components are isolated from Open Source components in terms of imports and dependencies.
2. **Modularity**: Develop a modular architecture where users can run only the AGPL-licensed parts without any reliance on proprietary code.
3. **Documentation**: Clearly document how to build and run a pure open-source version of `Automatisch`, providing guidelines for installation and configuration.

I appreciate this requires extra, unplanned work. I also appreciate the work already put in to `Automatisch`. However, this work is required in order to make Automatisch a genuine Open Source project.

**Comment by tnsai** [comment]
MIT licence would have been better. AGPL closes off a lot of use cases.

**Comment by inoop** [comment]
AGPL is a total no-go for commercial applications where the goal is to use Automatisch as an internal automation tool. A paid-for, hosted solution is also not an option because companies don't want their data on your servers. For these reasons, I'd seriously reconsider AGLP because it severely limits where Automatisch can run.
